codegen/src/androidMain/kotlin/com/freeletics/khonshu/codegen/internal/CollectAsState.kt

```diff
@@ -0,0 +1,20 @@
+package com.freeletics.khonshu.codegen.internal
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.State
+import androidx.compose.runtime.produceState
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.flowWithLifecycle
+import com.freeletics.khonshu.statemachine.StateMachine
+
+@Composable
+@InternalCodegenApi
+public fun <S : Any> StateMachine<S, *>.asComposeState(): State<S?> {
+    val lifecycleOwner = LocalLifecycleOwner.current
+    return produceState<S?>(initialValue = null, lifecycleOwner) {
+        state.flowWithLifecycle(lifecycleOwner.lifecycle, Lifecycle.State.RESUMED)
```

Comment:
I think `state` should be added to key list of `produceState`

Reply:
[b46b6bb](https://github.com/freeletics/khonshu/pull/668/commits/b46b6bb2bb3eac9731dc540d6fcb39e61f3b9ced) 
I've added a `remember` for the state as well because we can't assume that the `StateMachine` will return the same `Flow` on each call even if it's effectively the same `Flow` (e.g. it might do `channel.receiveAsFlow()` on each call).

Reply:
Nice, that makes sense 🙏